src/main/java/SearchEngine.java

```diff
@@ -0,0 +1,88 @@
+import strategies.impl.AllStrategyImpl;
+import strategies.impl.AnyStrategyImpl;
+import strategies.impl.NoneStrategyImpl;
+import strategies.SearchStrategy;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.Scanner;
+
+class SearchEngine {
+
+    private final FileReader fileReader;
+    private final Scanner scanner;
+    private final LineTokenizer lineTokenizer;
+
+
+    public SearchEngine(FileReader fileReader) {
+        this.fileReader = fileReader;
+        scanner = new Scanner(System.in);
+        lineTokenizer = new LineTokenizer();
+    }
+
+    public void run() throws IOException {
+        List<String> peopleList = fileReader.readFile();
+        while (true) {
+            printMenu();
+            String selection = scanner.nextLine();
+            switch (selection) {
+                case "1" -> find(peopleList, lineTokenizer.mapWords(peopleList));
+                case "2" -> printList(peopleList);
+                case "3" -> {
+                    scanner.close();
+                    System.out.println("\nBye!");
+                    return;
+                }
+                default -> System.out.println("\nIncorrect option! Try again.");
+            }
+        }
+    }
+
+    private void printMenu() {
+        System.out.println("""
+                === Menu ===
+                1. Find a person
+                2. Print all people
+                0. Exit""");
+    }
+
+    private void printList(List<String> list) {
+        System.out.println("\n=== List of people ===");
+        list.stream().map(String::new).forEach(s -> System.out.println(s.trim()));
```

Comment:
Please use each method of Stream from the new line. (That's a common rule when you work with Stream API)
e.g.
```
bars.stream()
   .map(..)
   .collect(..)
```
Why do you need ".map(String::new)", if you have a list of String as input?

Reply:
I don't need this, now deleted